### PR TITLE
GEODE-5674: Stop using random values for test ports

### PR DESCRIPTION
**Pull request by WireBaron**
It seems our test framework will fairly frequently have test failing due to
multiple users trying to bind the same port number.  This happens when the
different users happen to randomly generate the same port.  To keep this from
happening, this change will simply hand out the available ports consecutively
as the test runs.  For dunit tests, each child vm will be given a portion of
the overall range for their use.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java [hunk 1]
```diff
@@ -38,6 +34,23 @@
  * further calls to getRandomAvailablePort.
  */
 public class AvailablePortHelper {
+  static AtomicInteger currentMembershipPort;
```
Comment (kirklund) [inline geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java]: I recommend making these private final. You can still change the values of them in the static block.

File: geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java [hunk 2]
```diff
@@ -38,6 +34,23 @@
...
+  static AtomicInteger currentAvailablePort;
+
+  static {
```
Comment (kirklund) [inline geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java]: Wouldn't this static block be better as a method? You could still invoke the method from the static block.

Overall, it looks like this class would benefit from changing to be a singleton (I hate to say it) but at least that would be cleaner from a design perspective than having static variables and static blocks.

File: geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java [hunk 3]
```diff
@@ -147,16 +136,15 @@ private static void releaseKeepers(List<Keeper> keepers) {
...
-    }
-    int[] ports = new int[set.size()];
+    int[] ports = new int[count];
     int i = 0;
-    for (Iterator iter = set.iterator(); iter.hasNext();) {
-      ports[i] = ((Integer) iter.next()).intValue();
-      i++;
+    while (i < count) {
+      int port = getUniquePort(false, AvailablePort.SOCKET);
+      while (port % site != 0) {
```
[the remark this answers is not in the document]
Comment (galen-pivotal) [reply geode-junit/src/main/java/org/apache/geode/internal/AvailablePortHelper.java]: To be clear: this is the only comment whose response is blocking merging this; all the rest of it is basically exposition and I'll merge without it.

**Comment by WireBaron** [comment]
This change will fail any UDP connection as isPortKeepable will always fail on those.  That ends up breaking the single point of contact between the AvailablePortHelper and AvailablePort code which was fundamental to this change.  This is probably salvageable, but will require a rather fundamental rework.  Closing this PR for now.

**Pull request closed without merging**